api/distributor/distributor.proto

```diff
@@ -16,8 +18,18 @@ message DistributedRequest {
   repeated envoy.config.core.v3.Address services = 3 [(validate.rules).repeated .min_items = 1];
 }
 
+message DistributedResponseFragment {
```

Comment:
can we add a top-level explanation of the purpose of this object?

Also, looking at the code, I think that the intention here is that one of these is associated with every nighthawk service, and essentially operates as either:
1. We failed to connect to this service
2. Here's the service address.

Given that, is DistributedResponseFragment a good name here?  I don't love the name fragment here, because it implies to me that they are parts of a whole (often without independent meaning outside of that whole).

Reply:
Yes this is spot on; I'll think about a better name here. Also open to suggestions.

Reply:
The DistributorRequest references `services`, so for a name how about `DistributedServiceResponse` or just `DistributedResponse`?

I think you were using `Fragment` to indicate that it's not the full resultset, but I don't think that's necessary here, as long as we document the intent.

(Some APIs I have used in the past have referred to this as an AsyncResponse, but I'm not sure how conventional that is).

Reply:
Renamed in 5462844f. I went with `DistributedServiceResponse`, is `DistributedResponse` was already used in:

```proto
message DistributedResponse {
  repeated DistributedServiceResponse service_response = 1;
}
```